File: API/Dms.Domain/Entities/DanhMucMonTheThao.cs
using Dms.Domain.Common;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Dms.Domain.Entities
{
    [Table("DanhMucMonTheThao")]
    public class DanhMucMonTheThao : BaseEntity
    {
        [Required]
        [MaxLength(50)]
        public string Ma { get; set; } = string.Empty;

        [Required]
        [MaxLength(200)]
        public string Ten { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string? MoTa { get; set; }

        public bool TrangThai { get; set; } = true;

        public virtual ICollection<MonTheThao> MonTheThaos { get; set; } = new List<MonTheThao>();
    }
}

File: my-app/src/types/danhMucMonTheThao.ts
export interface DanhMucMonTheThao {
  id: number;
  ma: string;
  ten: string;
  moTa?: string;
  trangThai: boolean;
  soMonTheThao?: number;
  created?: string;
  lastModified?: string;
}

export interface CreateUpdateDanhMucMonTheThao {
  ma: string;
  ten: string;
  moTa?: string;
  trangThai: boolean;
}
